agent/consul/state/catalog.go

```diff
@@ -1942,10 +1950,52 @@ func (s *Store) CheckConnectServiceNodes(ws memdb.WatchSet, serviceName string,
 	return s.checkServiceNodes(ws, serviceName, true, entMeta)
 }
 
+// CheckIngressServiceNodes is used to query all nodes and checks for ingress
+// endpoints for a given service.
+func (s *Store) CheckIngressServiceNodes(ws memdb.WatchSet, serviceName string, entMeta *structs.EnterpriseMeta) (uint64, structs.CheckServiceNodes, error) {
+	tx := s.db.Txn(false)
+	defer tx.Abort()
+	maxIdx, nodes, watchChs, err := s.serviceGatewayNodes(tx, serviceName, structs.ServiceKindIngressGateway, entMeta)
+	if err != nil {
+		return 0, nil, fmt.Errorf("failed gateway nodes lookup: %v", err)
+	}
+
+	// TODO(ingress) : Deal with incorporating index from mapping table
+
+	// Watch list of gateway nodes for changes
+	for _, ch := range watchChs {
+		ws.Add(ch)
+	}
+
+	// De-dup service names to lookup
+	serviceNames := make(map[string]struct{})
```

Comment:
Does this need to be deduped by `ServiceID` instead to take namespaces into account in ent?

Reply:
Yep, you're right, pushed a commit and added a TODO to test this in enterprise. 👍